### DISCUSSION TITLE
Button Texture Duplicate

### DISCUSSION THREAD

**Discussion by Marving1023** [SFX-WoW/Masque#175] [Issues] [answered]
Hello,

I use this code with WeakAuras2 and Custom Tringer [PLAYER_ENTERING_WORLD]. 
Masque is disabled for WeakAuras2 but not for my frame
All works when I connect, but after /console reloadUI
I feel like the skins are stacked several times.

Is there something wrong with my code?

Thank you for your reply.

> 	local icon_region =  WeakAuras.regions[Core.Name].region
	local button_region = CreateFrame("Button", Core.Name, icon_region, "SecureActionButtonTemplate")

	-- #SKINS REGION
	local MSQGroup = MSQ:Group("WeakAuras", string.gsub(string.lower(Core.Name), "%s+", "_"))
	button_region:SetAllPoints()
	button_region:SetHighlightTexture([[Interface\Buttons\ButtonHilight-Square]])
	button_region:SetPushedTexture([[Interface\Buttons\UI-Quickslot-Depress]])
	MSQGroup:AddButton(button_region, button_region)

	-- # ACTION BUTTON
	button_region:RegisterForClicks("AnyUp")
	if Core.MacroCondition ~= nil then
		local spell_info_name = select(1, GetSpellInfo(Core.SpellId))
		local player_macrotext = Core.MacroCondition .. "\n/cast " .. spell_info_name
		button_region:SetAttribute("type", "macro")
		button_region:SetAttribute("macrotext", player_macrotext)
	elseif Core.Type == "spell" then
		local spell_info_name = select(1, GetSpellInfo(Core.SpellId))
		button_region:SetAttribute("type", Core.Type)
		button_region:SetAttribute(Core.Type, spell_info_name)
	elseif (Core.Type == "item") then
		local item_info_name = Core.Type .. ":" .. Core.SpellId
		button_region:SetAttribute("type", Core.Type)
		button_region:SetAttribute(Core.Type, item_info_name)
	end

	-- # EVENT ONENTER
	button_region:SetScript("OnEnter",
		function()
			GameTooltip:SetOwner(TooltipMover, "BOTTOM_RIGHT", 129, 18)--ElvUI
			if Core.Type == "spell" then
				GameTooltip:SetSpellByID(Core.SpellId)
			elseif Core.Type == "item" then
				GameTooltip:SetItemByID(Core.SpellId)
			end
			GameTooltip:Show()
		end)

	-- # EVENT ONLEAVE
	button_region:SetScript("OnLeave",
		function()
			GameTooltip:Hide()
		end)

	-- # VISIBILITY STATE
	if Core.VisibilityState ~= nil then
		RegisterStateDriver(icon_region, "visibility", Core.VisibilityState)
		RegisterStateDriver(button_region, "visibility", Core.VisibilityState)
	end

**Comment by Marving1023** [comment, marked as answer]
Resolved

WeakAuras2 runs the trigger several times at startup [PLAYER_ENTERING_WORLD] and duplicate my custom frame
https://github.com/WeakAuras/WeakAuras2/wiki/Custom-Triggers

**Comment by StormFX** [comment]
Sorry for the late response. Glad you figured out. I would have suggested checking for frame duplication, as Masque's storage system prevents duplicate skins.
